qemu/tests/x86_cpu_protection_key.py

```diff
@@ -0,0 +1,62 @@
+from virttest import cpu
+from virttest import env_process
+from virttest import error_context
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Get kernel src code from src rpm and run protection key tests.
+
+    1) Download src rpm from brew
+    2) Unpack src code and compile protection_keys.c
+    3) Run executable file 'protection_keys'
+    4) Check results
+
+    :param test:   QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env:    Dictionary with test environment.
+    """
+    unsupported_models = params.get("unsupported_models", "")
+    cpu_model = params.get("cpu_model")
+    if not cpu_model:
+        cpu_model = cpu.get_qemu_best_cpu_model(params)
+    if cpu_model in unsupported_models.split():
+        test.cancel("'%s' doesn't support this test case" % cpu_model)
+
+    params["start_vm"] = "yes"
+    vm_name = params['main_vm']
+    env_process.preprocess_vm(test, params, env, vm_name)
+
+    vm = env.get_vm(vm_name)
+    error_context.context("Try to log into guest", test.log.info)
+    session = vm.wait_for_login()
+
+    guest_dir = params["guest_dir"]
+    timeout = int(params["timeout"])
+    kernel_v = session.cmd_output("uname -r").strip()
+    mkdir_cmd = session.cmd('mkdir -p %s' % guest_dir)
+    src_rpm = "kernel-" + kernel_v[:-7] + ".src.rpm"
```

Comment:
`kernel_v[:-7]` could not always return the correct version. The string length of different architectures is not the same. Suggest use `.rsplit(".", 1)`

Reply:
> `kernel_v[:-7]` could not always return the correct version. The string length of different architectures is not the same. Suggest use `.rsplit(".", 1)`

updated and thanks.